runtime/drivers/registry.go

```diff
@@ -89,3 +108,47 @@ func (i *Instance) ResolveVariables() map[string]string {
 	}
 	return r
 }
+
+// Config resolves the current dynamic config properties for the instance.
+// See InstanceConfig for details.
+func (i *Instance) Config() (InstanceConfig, error) {
+	// Default config
+	res := InstanceConfig{
+		DownloadRowLimit:                  10_000,
+		PivotCellLimit:                    1_000_000,
+		InteractiveSQLRowLimit:            10_000,
+		StageChanges:                      true,
+		ModelDefaultMaterialize:           false,
+		ModelMaterializeDelaySeconds:      0,
+		AlertsDefaultStreamingRefreshCron: "*/10 * * * *", // Every 10 minutes
+	}
+
+	// Backwards compatibility: Use "__materialize_default" as alias for "rill.models.default_materialize".
+	if i.Variables != nil {
+		if v, ok := i.Variables["__materialize_default"]; ok {
+			if _, ok := i.Variables["rill.models.default_materialize"]; !ok {
+				i.Variables["rill.models.default_materialize"] = v
+			}
+		}
+	}
+
+	// Decode variables into res.
+	// Minor optimization: Instead of calling ResolveVariables (which allocates a new map), just call Decode on the underlying maps in the same order as in ResolveVariables.
```

Comment:
True, will just call `ResolveVariables` for now. Premature optimization :) 

Caching the result of `ResolveVariables` is unfortunately not completely trivial because the `Instance` struct is mutated in-place.